compiler-rt/test/asan/TestCases/asan_lsan_deadlock.cpp

```diff
@@ -0,0 +1,72 @@
+// Test for potential deadlock in LeakSanitizer+AddressSanitizer.
+// REQUIRES: leak-detection
+//
+// RUN: %clangxx_asan -O0 %s -o %t

```

Comment:
I assume this test will fail if we remove fix above?

Reply:
Due to the inability to precisely control the locking order between LSan and ASan, this test tends to fail with high probability only when the system is idle (in my environment).